src/GafferScene/CameraQuery.cpp

```diff
@@ -0,0 +1,551 @@
+					parameters->writable()[g_fStop] = new CompoundData( {
+						{ g_source, camera->parameters().count( g_fStop ) ? g_sourceCamera : g_sourceFallback },
+						{ g_value, new FloatData( camera->getFStop() ) }
+					} );
+
+					// SceneAlgo::applyCameraGlobals outputs an absolute shutter value that is dependent on
+					// the render:transformBlur and render:deformationBlur options, so we instead
+					// perform our own manual fallback through the relative shutter values.
+					V2fDataPtr shutterValue;
+					StringDataPtr shutterSource;
+					if( camera->hasShutter() )
+					{
+						shutterSource = g_sourceCamera;
+						shutterValue = new V2fData( camera->getShutter() );
+					}
+					else if( auto s = scenePlug()->globals()->member<V2fData>( g_renderShutterOption ) )
+					{
+						shutterSource = g_sourceGlobals;
+						shutterValue = s->copy();
+					}
+					else
+					{
+						shutterSource = g_sourceFallback;
+						shutterValue = new V2fData( camera->getShutter() );
+					}
+					parameters->writable()[g_shutter] = new CompoundData( {
+						{ g_source, shutterSource },
+						{ g_value, shutterValue }
+					} );
+
+					// Fall back to default values for all other registered camera parameters.
+					for( const auto &target : Metadata::targetsWithMetadata( "parameter:camera:*", g_defaultValue ) )
+					{
+						const std::string name = target.string().substr( 17 );
+						if( !parameters->readable().count( name ) )
+						{
+							parameters->writable()[name] = new CompoundData( {
+								{ g_source, g_sourceFallback },
+								{ g_value, Metadata::value( target, g_defaultValue )->copy() }
+							} );
+						}
+					}
+
+					// Create virtual parameters, for convenience these are pre-computed
+					// from the camera so they can be later accessed like a regular parameter.
+					const Imath::V2f aperture = cameraWithGlobals->getAperture();
+					parameters->writable()["apertureAspectRatio"] = new CompoundData( {
+						{ g_source, g_sourceCamera },
+						{ g_value, new FloatData( aperture[0] / aperture[1] ) }
+					} );
+					parameters->writable()["fieldOfView"] = new CompoundData( {
+						{ g_source, g_sourceCamera },
+						{ g_value, new V2fData( cameraWithGlobals->calculateFieldOfView() ) }
+					} );
+					parameters->writable()["frustum"] = new CompoundData( {
+						{ g_source, g_sourceCamera },
+						{ g_value, new Box2fData( cameraWithGlobals->frustum() ) }
+					} );
+				}
+			}
+		}
+
+		static_cast<AtomicCompoundDataPlug *>( output )->setValue( parameters );
+		return;
+	}
+	else if( outPlug()->isAncestorOf( output ) )
+	{
+		const ValuePlug *oPlug = outPlug( output );
+
+		const IECore::ConstCompoundDataPtr parameters = internalParametersPlug()->getValue();
+		assert( parameters );
+
+		if( output == oPlug->getChild( g_sourcePlugIndex ) )
+		{
+			const std::string parameterName = queryPlug( output )->getValue();
+			std::string source = g_sourceNone->readable();
+			if( parameterName.size() )
```

Comment:
This `parameterName` size check is common to both `source` and `value` branches. I wonder if it's worth lifting it out and doing an early return with `setToDefault()` before we hit the meat of the logic, reducing the overall nesting level by 1? (We'd need to make the default for the `source` plug be "none" for this to work).

Reply:
Thanks, that does simplify things a bit, done in 94c12bb8d90c63b33920edb43ffd0082146b3c19